### ISSUE TITLE
Typescript

### ISSUE THREAD

**Issue by bushaev-denis** [redaxle/knex-react-native-sqlite#1]
You planned add ts support for your library?

**Comment by dez64ru** [comment]
I think this is enough to support ts

/index.d.ts
```
declare module 'knex-react-native-sqlite' {
  import knex from './src/knex'
  import RNSqliteDialect from './src/dialect'

  export {knex, RNSqliteDialect}

  export default RNSqliteDialect
}
```

**Comment by redaxle** [comment]
@dez64ru saw the code suggestion, did you want to submit a pull request? Definitely something we're interested in adding

**Comment by bushaev-denis** [comment]
@redaxle sorry, i not enough know ts for this :(
right now i just do this

```tsx
import {Knex} from 'knex'
import { knex, RNSqliteDialect } from 'knex-react-native-sqlite'
import {Item} from './types'
const dbVersion = 1

export const openDB = async (): Promise<Knex> => {
  const db = knex({
    client:     RNSqliteDialect,
    connection: {
      location: 'default',
      name:     'database.db'
    }
  }) as Knex
};
```

**Comment by bushaev-denis** [comment]
I cant push into your repo, you need turn on PRs i think

**Comment by redaxle** [comment]
@dez64ru PRs are supported. What you need to do is clone this repo then commit and push a branch then you can create a PR. I didn't get a chance to test with Typescript yet but if you submit the PR then I can just test what you have.

**Comment by bushaev-denis** [comment]
@redaxle I should create my own branch or commit into main?

**Comment by bushaev-denis** [comment]
![Screenshot 2022-06-27 at 17 33 12](https://user-images.githubusercontent.com/32451886/175954118-d4e1964f-0936-4805-96e6-456c5058d496.png)
Ive test with main and self-created branch. Still has error :(

**Comment by redaxle** [comment]
@dez64ru ah, that's the problem. You are trying to push to the redaxle repo. You want to first fork, then clone and create a new branch, push that branch to your remote/fork (should be something like github.com/dez64ru/knex-react-native-sqlite) then you can create a PR on the fork which will take you back to this repo. Let me know if you still have problems.

**Comment by bushaev-denis** [comment]
@redaxle thanks for help. Its my first time open source PR :)

**Comment by redaxle** [comment]
@dez64ru thanks again, now resolved by #2
